src/config/vars.rs

```diff
@@ -4,7 +4,20 @@ use std::collections::HashMap;
 /// (result, warnings)
 pub fn interpolate(input: &str, vars: &HashMap<String, String>) -> (String, Vec<String>) {
     let mut warnings = Vec::new();
-    let re = Regex::new(r"\$\$|\$(\w+)|\$\{(\w+)(?::-([^}]+)?)?\}").unwrap();
+    // Environment variable names can have any characters from the Portable Character Set other
+    // than NUL.  However, for Vector's intepolation, we are closer to what a shell supports which
+    // is solely of uppercase letters, digits, and the '_' (that is, the `[:word:]` regex class).
+    // In addition to these characters, we allow `.` as this commonly appears in environment
+    // variable names when they come from a Java properties file.
+    //
+    // https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html
+    let re = Regex::new(
+        r"(?x)
+        \$\$|
+        \$([[:word:]-.]+)|
+        \$\{([[:word:]-.]+)(?::-([^}]+)?)?\}",
+    )
```

Comment:
Mostly just trusting you on the regex here, but if it's working out in the tests 🤷